Model/Gateway.php

```diff
@@ -129,31 +130,46 @@ public function __construct(
     /**
      * Performs a purchase against the FatZebra gateway with a tokenized credit card
      *
-     * @param mixed $token
-     * @param mixed $amount
-     * @param mixed $reference
-     * @param mixed $cvv
-     * @param mixed $fraud_data
+     * @param $pmntsToken
+     * @param $amount
+     * @param $reference
+     * @param $cvv
+     * @param $fraud_data
+     * @param $paymentType
      * @return mixed
      * @throws \Exception
      */
-    public function tokenPurchase($token, $amount, $reference, $cvv = null, $fraud_data = null)
+    public function tokenPurchase($pmntsToken, $amount, $reference, $cvv = null, $fraud_data = null, $paymentType=null)
     {
+        $payload = [];
         $customer_ip = $this->getCustomerIp();
-
         if (function_exists('bcmul')) {
             $int_amount = (int)bcmul((string)$amount, "100");
         } else {
             $multiplied = round($amount * 100);
             $int_amount = (int)$multiplied;
         }
-        $payload = [
-            "customer_ip" => $customer_ip,
-            "card_token" => $token,
-            "cvv" => $cvv,
-            "amount" => $int_amount,
-            "reference" => $reference
+
+        if ($paymentType && $paymentType == self::PMNTS_GOOGLE_PAYMENT_METHOD_CODE) {
+            $token = json_decode($pmntsToken, true);
```

Comment:
Since this is a public function can you check after decode the values converting to the expected array?